spring-cloud-huawei-nacos/src/main/java/com/huaweicloud/nacos/authentication/NacosInstanceService.java

```diff
@@ -0,0 +1,125 @@
+/*
+
+ * Copyright (C) 2020-2022 Huawei Technologies Co., Ltd. All rights reserved.
+
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.huaweicloud.nacos.authentication;
+
+import com.alibaba.cloud.nacos.NacosDiscoveryProperties;
+import com.alibaba.cloud.nacos.discovery.NacosServiceDiscovery;
+import com.alibaba.cloud.nacos.registry.NacosRegistration;
+import com.alibaba.nacos.api.exception.NacosException;
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.util.concurrent.UncheckedExecutionException;
+import com.huaweicloud.common.disovery.InstanceIDAdapter;
+import com.huaweicloud.governance.authentication.Const;
+import com.huaweicloud.governance.authentication.MicroserviceInstanceService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.cloud.client.ServiceInstance;
+
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+public class NacosInstanceService implements MicroserviceInstanceService {
+    private static final Logger LOGGER = LoggerFactory.getLogger(NacosInstanceService.class);
+
+    private final NacosDiscoveryProperties properties;
+
+    private final NacosServiceDiscovery serviceDiscovery;
+
+    private final NacosRegistration registration;
+
+    private String publickey = "";
+
+    public NacosInstanceService(NacosDiscoveryProperties properties, NacosRegistration registration,
+        NacosServiceDiscovery serviceDiscovery) {
+        this.properties = properties;
+        this.registration = registration;
+        this.serviceDiscovery = serviceDiscovery;
+    }
+
+    @Override
+    public void setPublickey(String publicKeyEncoded) {
+        publickey = publicKeyEncoded;
+        properties.getMetadata().put(Const.INSTANCE_PUBKEY_PRO, publicKeyEncoded);
+    }
+
+    @Override
+    public String getInstanceId() {
+        ServiceInstance serviceInstance = null;
+        try {
+            serviceInstance = serviceDiscovery.getInstances(registration.getServiceId())
+                    .stream()
+                    .filter(instance -> instance.getMetadata().containsValue(publickey))
+                    .findAny()
+                    .get();
+        } catch (NacosException e) {
+            throw new RuntimeException(e);
+        }
+        return InstanceIDAdapter.instanceId(serviceInstance);
```

Comment:
Can replaced with 'return InstanceIDAdapter.instanceId(registration);' ?

Reply:
changed to InstanceIDAdapter.instanceId(registration);